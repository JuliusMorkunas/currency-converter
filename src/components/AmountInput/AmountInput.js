import React from 'react';
import PropTypes from 'prop-types';
import s from './AmountInput.module.scss';
import Label from '../Label';

const AmountInput = ({
  label = '',
  currency = '',
  defaultValue,
  value,
  onChange,
  onKeyDown,
  autoFocus,
}) => {
  const id = `amount-${label.replace(/\s/g, '').toLowerCase()}`;
  return (
    <div className={s.Container}>
      <Label htmlFor={id}>{label}</Label>
      <input
        type="text"
        name={id}
        id={id}
        defaultValue={defaultValue}
        value={value}
        className={s.Input}
        onChange={onChange}
        onKeyDown={onKeyDown}
        autoFocus={autoFocus}
      />
      <span className={s.Currency}>{currency}</span>
    </div>
  );
};

AmountInput.propTypes = {
  label: PropTypes.string,
  currency: PropTypes.string,
  defaultValue: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
  autoFocus: PropTypes.bool,
};

export default AmountInput;
