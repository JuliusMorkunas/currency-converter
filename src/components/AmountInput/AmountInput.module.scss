@import '../../variables';

.Container {
  font-family: Arial, sans-serif;
  position: relative;
}

.Input {
  font-size: 29px;
  font-weight: 500;
  width: 100%;
  height: 42px;
  border: none;
  border-bottom: 1px solid $c-blueish-border;
  outline: none;
}

.Currency {
  font-size: 16px;
  line-height: 40px;
  position: absolute;
  right: 1px;
  bottom: 1px;
  height: 40px;
  padding-left: 10px;
  text-transform: uppercase;
  color: $c-grey;
  background-color: $c-white;
}
