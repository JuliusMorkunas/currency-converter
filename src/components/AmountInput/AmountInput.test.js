import React from 'react';
import { render } from '@testing-library/react';
import AmountInput from './AmountInput';

it('AmountInput should match snapshot', () => {
  const { asFragment } = render(<AmountInput />);
  expect(asFragment()).toMatchSnapshot();
});
