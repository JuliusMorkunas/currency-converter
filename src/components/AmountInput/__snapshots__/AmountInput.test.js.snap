// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container"
  >
    <label
      class="Label"
      for="amount-"
    >
      :
    </label>
    <input
      class="Input"
      id="amount-"
      name="amount-"
      type="text"
      value=""
    />
    <span
      class="Currency"
    />
  </div>
</DocumentFragment>
`;
